model.py

```diff
@@ -10573,6 +10573,17 @@ def lookup_hook():
             return True
         return False
 
+    def can_see_collection(self, collection):
+        for library in collection.libraries:
+            if self.is_librarian(library):
+                return True
+        if not collection.libraries:
+            # If the collection's not associated with any libraries, only system
+            # admins can see it.
+            if self.is_system_admin():
+                return True
```

Comment:
This is fine, but it would be a little simpler with the same results if you checked is_system_admin right off.